exotations/task_maps/task_map/src/Point2Line.cpp

```diff
@@ -85,15 +85,114 @@ void Point2Line::update(Eigen::VectorXdRefConst x, Eigen::VectorXdRef phi, Eigen
         phi(i) = dv.norm();
         for (int j = 0; j < J.cols(); j++)
         {
-            J(i, j) = -dv.dot(Eigen::Map<const Eigen::Vector3d>(Kinematics.J[i].getColumn(j).vel.data)) / dv.norm();
+            if (phi(i) > 1e-9)
+            {
+                J(i, j) = -(dv.normalized()).dot(Eigen::Map<const Eigen::Vector3d>(Kinematics.J[i].getColumn(j).vel.data));
```

Comment:
This is not the correct Jacobian if all three points are attached different robot links!
You need to apply chain rule on the function used to calculate `phi`.

Reply:
The start and end points of the line are defined in the same frame (the base frame, "Base"). And the `Kinematics` provides me with the link point in the base frame.

The `Distance` task map minimises the point distances the same way:
https://github.com/ipab-slmc/exotica/blob/e2bb512d40174ca2078f8eacbdef6439ca82730a/exotations/task_maps/task_map/src/Distance.cpp#L62

Am I missing something here?